### ISSUE TITLE
```tf.keras.callbacks.EarlyStopping``` doesn't set correct mode when ```monitor="val_auc",mode="auto"```

### ISSUE THREAD

**Issue by DachuanZhao** [tensorflow/tensorflow#44509]
**Describe the current behavior**
[https://colab.research.google.com/drive/1y-gYAIMHYokd9K-OUDXuD3PGTh4qE1xN?usp=sharing](https://colab.research.google.com/drive/1y-gYAIMHYokd9K-OUDXuD3PGTh4qE1xN?usp=sharing)
```
early_stopping = tf.keras.callbacks.EarlyStopping(
    monitor='val_auc',
    verbose=1,
    )
```
is equivalent to
```
early_stopping = tf.keras.callbacks.EarlyStopping(
    monitor='val_auc',
    verbose=1,
    mode="min"
    )
```
**Describe the expected behavior**
It should be equivalent to
```
early_stopping = tf.keras.callbacks.EarlyStopping(
    monitor='val_auc',
    verbose=1,
    mode="max"
    )
```

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with [TF v2.3](https://colab.research.google.com/gist/amahendrakar/afcaddf04d01641d4cbacb65b36ea51e/44509.ipynb) and [TF-nightly](https://colab.research.google.com/gist/amahendrakar/2a13236cc5f15233590e311ae622ef0f/44509-tf-nightly.ipynb). Please find the attached gist. Thanks!

**Comment by ymodak** [comment]
https://github.com/tensorflow/tensorflow/blob/fcc4b966f1265f466e82617020af93670141b009/tensorflow/python/keras/callbacks.py#L1640-L1643

Looks like for `acc` the `mode='max'` and for monitoring other quantities it is set to 'min'.
I think you have to manually set the `mode='max'` here for your case.

**Comment by DachuanZhao** [comment]
> https://github.com/tensorflow/tensorflow/blob/fcc4b966f1265f466e82617020af93670141b009/tensorflow/python/keras/callbacks.py#L1640-L1643
> 
> Looks like for `acc` the `mode='max'` and for monitoring other quantities it is set to `'min'`.
> I think you have to manually set the `mode='max'` here for your case.

ok，but why not change 
```Python
      if 'acc' in self.monitor:
        self.monitor_op = np.greater
      else:
        self.monitor_op = np.less
```
to
```Python
      if 'acc' in self.monitor or 'auc' in self.monitor:
        self.monitor_op = np.greater
      else:
        self.monitor_op = np.less
```
?

**Comment by ymodak** [comment]
This is fixed with tf nightly version `2.5.0-dev20201029`  and also with TF 2.4
('val_acc' is monitored for max value)
See [gist](https://colab.research.google.com/gist/ymodak/f7925391d77c532b56795fe57173fbad/44509-tf-nightly.ipynb)

**Comment by DachuanZhao** [comment]
> This is fixed with tf nightly version `2.5.0-dev20201029` and also with TF 2.4
> (`'val_acc'` is monitored for max value)
> See [gist](https://colab.research.google.com/gist/ymodak/f7925391d77c532b56795fe57173fbad/44509-tf-nightly.ipynb) for reference.

I don't think you fix it . In fact , it stops at the first epoch because this [issue](https://github.com/tensorflow/tensorflow/issues/49227) . Since keras is moved to its own repo , I will try to raise a PR to solve it .
